### PR TITLE
use new pulpcore repos for katello 3.18+

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: roles/forklift_versions/library/forklift_versions.py [hunk 1]
```diff
@@ -87,6 +87,8 @@ def main():
                         'pulp_repositories_version': version['pulp'],
                         'puppet_repositories_version': version['puppet'],
                         }
+                if 'pulpcore' in version:
```
Comment (evgeni) [inline roles/forklift_versions/library/forklift_versions.py:90]: this is separate, so the var is undefined for versions that don't use the split pulpcore

File: vagrant/config/versions.yaml [hunk 2]
```diff
@@ -90,6 +90,7 @@ installers:
   - foreman: 'nightly'
     katello: 'nightly'
     pulp: '2.21'
+    pulpcore: '3.4'
```
Comment (evgeni) [inline vagrant/config/versions.yaml:93]: same q @jlsherrill  ;)

**Review by ekohl — COMMENTED** [review]
Does it make sense to define a `pulpcore_repositories` role and include that if needed?

**Comment by evgeni** [comment]
I considered that, but thought it's only useful when one wants to use pulpcore without Katello, which is out of scope for this repo?

File: roles/katello_repositories/tasks/staging_repos.yml [hunk 3]
```diff
@@ -53,7 +53,20 @@
...
-    - katello_repositories_version == "nightly" or katello_repositories_version is version('3.15', '>=')
+    - katello_repositories_version != 'nightly'
+    - katello_repositories_version is version('3.15', '>=')
+    - katello_repositories_version is version('3.17', '<=')
+
+- name: 'Add Pulpcore repository'
+  yum_repository:
+    name: pulpcore-repository
+    description: Pulpcore repository for Katello {{ katello_repositories_version }}
+    baseurl: "https://yum.theforeman.org/pulpcore/{{ katello_repositories_pulpcore_version }}/el{{ ansible_distribution_major_version }}/x86_64/"
```
Comment (ekohl) [inline roles/katello_repositories/tasks/staging_repos.yml:64]: This doesn't look like the staging URL Is that intentional? Should there be some `pulpcore-release` package inside that repo itself?

File: roles/katello_repositories/tasks/staging_repos.yml [hunk 4]
```diff
@@ -53,7 +53,20 @@
...
+    gpgcheck: no
```
Comment (ekohl) [inline roles/katello_repositories/tasks/staging_repos.yml:65]: Any plans to GPG sign pulpcore releases?

File: vagrant/config/versions.yaml (hunk 2 above)
Comment (jlsherrill) [reply vagrant/config/versions.yaml:93]: 3.6 all the way

File: roles/katello_repositories/tasks/staging_repos.yml (hunk 3 above)
Comment (evgeni) [reply roles/katello_repositories/tasks/staging_repos.yml:64]: There is no staging as such (well, besides repoclosure). I can add such a package, yeah.

File: roles/katello_repositories/tasks/staging_repos.yml (hunk 4 above)
Comment (evgeni) [reply roles/katello_repositories/tasks/staging_repos.yml:65]: Yes, it's on my todo for this week.

**Comment by evgeni** [comment]
@ekohl I've posted the alternative with a separate `pulpcore_repositories` role in https://github.com/theforeman/forklift/pull/1199, which one do you like better?

**Comment by evgeni** [comment]
closing in favor of #1199

**Pull request closed without merging**